src/time_manager.cpp

```diff
@@ -0,0 +1,40 @@
+/**
+ * @file
+ * @brief     CPP file for the TimeManager class to control the DS3231 RTC
+ * module
+ * @author    Jasper Smienk
+ * @license   MIT License
+ */
+
+#include "time_manager.hpp"
+
+namespace Time {
+TimeManager::TimeManager(hwlib::pin_oc &scl, hwlib::pin_oc &sda)
+    : scl(scl), sda(sda), realTimeClock(hwlib::i2c_bus_bit_banged_scl_sda(scl, sda)) {
```

Comment:
It is preferred to instantiate hwlib::i2c_bus_bit_banged_scl_sda outside of the class definitions. Instead a hwlib::i2c_bus& should be used internally so that the class does not care about implementation nor does it select one itself. (Useful for unit testing)